README.md

```diff
@@ -134,6 +134,49 @@ becomes
       extra-lib-dirs:
         lib/darwin
 
+Multiple conditionals must fall under one `when`, for example:
+
+    when:
+      - condition: os(darwin)
+        extra-lib-dirs: lib/darwin
+      - condition: os(linux)
+        extra-lib-dirs: /usr/lib/x64
+
+becomes
+
+    if os(darwin)
+      extra-lib-dirs:
+        lib/darwin
+    if os(linux)
+      extra-lib-dirs:
+        /usr/lib/x64
+
+If conditionals are declared under different `when` clauses only the last one
+will make it to the `.cabal` file:
```

Comment:
You're right that this behaviour is on the YAML side of things (and is actually wrong according to the spec: http://yaml.org/spec/1.1/#id932806), but I find it very useful that it behaves as it does, and would actually propose to add tests for this behaviour and file bug reports to https://hackage.haskell.org/package/yaml if the behaviour ever changes. Here are two examples why:

Aliases with overriding:

    executables:
        strava-gear-cmdline:
            &default_executable
            main: strava-gear-cmdline.hs
            dependencies: [strava-gear]
            ghc-options:
                - -threaded

        strava-gear-web:
            <<: *default_executable
            main: strava-gear-web.hs

Predefined aliases:

    executables:
        &default_executable
        dependencies: [strava-gear]
        ghc-options:
            - -threaded

    executables:
        strava-gear-cmdline:
            <<: *default_executable
            main: strava-gear-cmdline.hs

        strava-gear-web:
            <<: *default_executable
            main: strava-gear-web.hs

The second example is a massive hack, but I'm not aware of any other way to do this given that hpack insists on printing warnings for unused fields even if I prefix them with underscores.